libs/libc/machine/arch_atomic.c

```diff
@@ -40,23 +46,23 @@
   void weak_function CONCATENATE(fn, n) (FAR volatile void *ptr,   \
                                          type value, int memorder) \
   {                                                                \
-    irqstate_t irqstate = spin_lock_irqsave(NULL);                 \
+    irqstate_t irqstate = spin_lock_irqsave(&g_atomic_lock);       \
```

Comment:
If recursion support is removed, calling `*_wo_note()` version is more safer.

Reply:
ok